src/inabox/inabox-viewport.js

```diff
@@ -174,6 +174,27 @@ export class ViewportBindingInabox {
         });
   }
 
+  /** @override */
+  getElementOpacity(el) {
```

Comment:
To not blow the size of AMP runtime, we'd like this to be moved to amp-analytics extension.

So no change needed to resource.js, viewport-*.js.
You can probably introduce a new file opacity.js under `amp-analytics/0.1/` and have all the logic there.